### ISSUE TITLE
1.3.0 - Not compatible with Contao 4.13

### ISSUE THREAD

**Issue by xx81** [terminal42/contao-shortlink#24]
Hi,

after the newest commit/update  ( https://github.com/terminal42/contao-shortlink/commit/77075d228681576a2810f423934f690595a348ba ) with version 1.3.0

we are getting the following error and contao 4.13 is stop working:

Terminal42\ShortlinkBundle\Picker\ShortlinkPickerProvider::supportsContext(string $context): bool must be compatible with Contao\CoreBundle\Picker\PickerProviderInterface::supportsContext($context)

I am guess that the interface in Version 4.13 is different from your code:
https://github.com/contao/contao/blob/4.13/core-bundle/src/Picker/PickerProviderInterface.php#L47

In Version 5.x it looks like this:
https://github.com/contao/contao/blob/5.x/core-bundle/src/Picker/PickerProviderInterface.php#L37

The type decleration of the arguments added in Version 5.0 of contao but not in Version 4.13

So we make a downgrade to 1.2.x ...

**Comment by aschempp** [comment]
@terminal42/pilots is there a way we can implement both?

**Comment by Toflar** [comment]
Probably by just dropping the type hints. But you could also simply make 5.3 the minimum requirement and work with 2 versions.

**Comment by xx81** [comment]
Another Idea - I know is the following:

https://github.com/terminal42/contao-shortlink/blob/77075d228681576a2810f423934f690595a348ba/src/Picker/ShortlinkPickerProvider.php#L3

What about to change

`declare(strict_types=1);`

to

`declare(strict_types=0);`

I know, that this i not the best solution, but one option for this specific file.

But when doing this, i am not quite sure, if you get problems with Easy Coding Standards auto injecting Head-Comments or something like this.

**Comment by fritzmg** [comment]
@Toflar removing the type hints will make it incompatible with the Contao 5 interface signature.

@xx81 that won't work either as `strict_types` has nothing to do with the issue (incompatible interface signatures).

The only solution I think is to revert https://github.com/terminal42/contao-shortlink/pull/23, release as `1.3.1` and then re-apply https://github.com/terminal42/contao-shortlink/pull/23 - but this time without https://github.com/terminal42/contao-shortlink/pull/23/commits/41c6ba999a3c460642b982b7a15ffd4bdca6f9e1.

**Comment by aschempp** [comment]
Not sure that's true. Apparently, I don't need to implement the type defined in an interface: https://3v4l.org/2KJPu 🤔

**Comment by aschempp** [comment]
@xx81 can you check if the issue is fixed in `dev-main`?

**Comment by fritzmg** [comment]
> Not sure that's true. Apparently, I don't need to implement the type defined in an interface: https://3v4l.org/2KJPu 🤔

True, the return type can be more specific.

> @xx81 can you check if the issue is fixed in `dev-main`?

Unfortunately there is still another issue:

```
Fatal error: Declaration of Terminal42\ShortlinkBundle\Routing\RouteProvider::getRouteByName(string $name): Symfony\Component\Routing\Route 
must be compatible with Symfony\Cmf\Component\Routing\RouteProviderInterface::getRouteByName($name)
```

**Comment by fritzmg** [comment]
Fixing the `RouteProvider` in the same way (removing the parameter types from `getRouteByName` and `getRouteByNames`) works as well though 👍

**Comment by aschempp** [comment]
Can you please test dev-main again?

**Comment by aschempp** [comment]
Released as 1.3.1 then 👍
